controller/src/main/java/io/pravega/controller/server/eventProcessor/requesthandlers/AbstractRequestProcessor.java

```diff
@@ -0,0 +1,179 @@
+        super(executor);
+        Preconditions.checkNotNull(streamMetadataStore);
+        this.streamMetadataStore = streamMetadataStore;
+    }
+
+    public String getProcessorName() {
+        return this.getClass().getSimpleName();
+    }
+
+    @Override
+    public CompletableFuture<Void> processEvent(ControllerEvent controllerEvent) {
+        return controllerEvent.process(this);
+    }
+
+    @Override
+    public CompletableFuture<Void> processAbortTxnRequest(AbortEvent abortEvent) {
+        return Futures.failedFuture(new RequestUnsupportedException("Request Unsupported"));
+    }
+
+    @Override
+    public CompletableFuture<Void> processCommitTxnRequest(CommitEvent commitEvent) {
+        return Futures.failedFuture(new RequestUnsupportedException("Request Unsupported"));
+    }
+
+    @Override
+    public CompletableFuture<Void> processAutoScaleRequest(AutoScaleEvent autoScaleEvent) {
+        return Futures.failedFuture(new RequestUnsupportedException("Request Unsupported"));
+    }
+
+    @Override
+    public CompletableFuture<Void> processScaleOpRequest(ScaleOpEvent scaleOpEvent) {
+        return Futures.failedFuture(new RequestUnsupportedException("Request Unsupported"));
+    }
+
+    @Override
+    public CompletableFuture<Void> processUpdateStream(UpdateStreamEvent updateStreamEvent) {
+        return Futures.failedFuture(new RequestUnsupportedException("Request Unsupported"));
+    }
+
+    @Override
+    public CompletableFuture<Void> processTruncateStream(TruncateStreamEvent truncateStreamEvent) {
+        return Futures.failedFuture(new RequestUnsupportedException("Request Unsupported"));
+    }
+
+    @Override
+    public CompletableFuture<Void> processSealStream(SealStreamEvent sealStreamEvent) {
+        return Futures.failedFuture(new RequestUnsupportedException("Request Unsupported"));
+    }
+
+    @Override
+    public CompletableFuture<Void> processDeleteStream(DeleteStreamEvent deleteStreamEvent) {
+        return Futures.failedFuture(new RequestUnsupportedException("Request Unsupported"));
+    }
+
+    protected <T extends ControllerEvent> CompletableFuture<Void> withCompletion(StreamTask<T> task, T event, String scope, String stream,
+                                                                                 Predicate<Throwable> writeBackPredicate) {
+        Preconditions.checkNotNull(task);
+        Preconditions.checkNotNull(event);
+        Preconditions.checkNotNull(scope);
+        Preconditions.checkNotNull(stream);
+        Preconditions.checkNotNull(writeBackPredicate);
+        CompletableFuture<Void> resultFuture = new CompletableFuture<>();
+
+        OperationContext context = streamMetadataStore.createContext(scope, stream);
+        suppressException(streamMetadataStore.getWaitingRequestProcessor(scope, stream, context, executor), null,
+                "Exception while trying to fetch waiting request. Logged and ignored.")
```

Comment:
we are logging an ignoring because the failure in this call is not deal breaker. It will just mean we dont honor fairness but try to take the processing nevertheless.

Reply:
Ok, but the logging should be at least warn level.